src/net/stackerdb/bits.rs

```diff
@@ -0,0 +1,123 @@
+// Copyright (C) 2013-2020 Blockstack PBC, a public benefit corporation
+// Copyright (C) 2020-2023 Stacks Open Internet Foundation
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+/// This module contains methods for interacting with the data contained within the messages
+use crate::net::{
+    Error as net_error, StackerDBChunkData, StackerDBChunkInvData, StackerDBGetChunkData,
+    StackerDBGetChunkInvData,
+};
+
+use stacks_common::util::hash::{Hash160, Sha512Trunc256Sum};
+
+use stacks_common::types::chainstate::{ConsensusHash, StacksAddress, StacksPrivateKey};
+
+use stacks_common::types::PrivateKey;
+
+use crate::net::stackerdb::ChunkMetadata;
+
+use crate::chainstate::stacks::StacksPublicKey;
+
+use stacks_common::util::secp256k1::MessageSignature;
+
+impl ChunkMetadata {
+    /// Get the digest to sign that authenticates this chunk data and metadata
+    fn sighash(&self) -> Sha512Trunc256Sum {
+        let mut bytes = vec![];
+        bytes.extend_from_slice(&self.rc_consensus_hash.0);
+        bytes.extend_from_slice(&self.chunk_id.to_be_bytes());
+        bytes.extend_from_slice(&self.chunk_version.to_be_bytes());
+        bytes.extend_from_slice(&self.data_hash.0);
+
+        Sha512Trunc256Sum::from_data(&bytes)
+    }
+
+    /// Sign this chunk metadata, committing to rc_consensus_hash, chunk_id, chunk_version, and
+    /// data_hash.
```

Comment:
Can you add a note to the rustdoc that the signature will be stored in `self.signature`?

Reply:
Yup